### PR TITLE
Fix docstring references in the manual

### LINKED ISSUE #2232: first/last/etc. documentation problem
http://juliadata.github.io/DataFrames.jl/stable/lib/types/

Below the "Common methods" header, some of the links are broken.
* first
* last
* unique
perhaps more.

### PR DESCRIPTION
**Pull request by mortenpi**
Try to fix #2232.

### CODE REVIEW

**Comment by mortenpi** [comment]
Ok, this should now actually resolve all the docstring referencing issues:

* Remove references to docstring that don't exist from the manual
* Remove at-refs to functions that don't have (DataFrames-specific) docstrings anymore
* Add in all the docstrings that exists, but were not included in the manual.

I also enabled `strict = true` for `makedocs` -- this should fail the documentation build if a PR introduces some new referencing problems.

File: docs/src/lib/functions.md [hunk 1]
```diff
@@ -80,3 +78,26 @@ transform!
...
+parent
+issorted
+getmaxwidths
+getprintedwidth
+gennames
+getchunkbounds
+showrowindices
+compacttype
+showrows
+```
```
Comment (mortenpi) [inline docs/src/lib/functions.md:103]: These are all the docstrings that were not included in the manual at the moment. These should be put into their appropriate sections in the manual before merging this, but I would prefer to leave this to someone who is more familiar with the overall structure of the package :slightly_smiling_face:

Comment (bkamins) [reply docs/src/lib/functions.md:103]: Thank you for doing this. Are we allowed to exclude the unexported functions from this list? (e.g. `ourshow` or `gennames`)

Comment (mortenpi) [reply docs/src/lib/functions.md:101]: They can't be completely removed from the manual, without Documenter complaining, but I would propose having a (hidden) internals page to get around this (3885ff3ad14ff7be3269a64b41ebd584c753f3db).

Comment (bkamins) [reply docs/src/lib/functions.md:101]: OK - thx.

File: docs/src/lib/functions.md [hunk 2]
```diff
@@ -80,3 +78,24 @@ transform!
...
+sortperm
+pairs
+parent
+issorted
+getmaxwidths
+getprintedwidth
+getchunkbounds
+showrowindices
+compacttype
+showrows
```
Comment (bkamins) [inline docs/src/lib/functions.md:100]:
```suggestion
first
last
unique
propertynames
similar
sortperm
pairs
parent
issorted
```

File: docs/src/lib/internals.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# Internals
+
+!!! warning "Internal API"
+
+    The functions, methods and types listed on this page are internal to DataFrames and are
+    **not considered to be part of the public API**.
+
+```@docs
+ourshow
+gennames
```
Comment (bkamins) [inline docs/src/lib/internals.md:14]:
```suggestion
compacttype
gennames
getchunkbounds
getmaxwidths
getprintedwidth
ourshow
ourstrwidth
showrowindices
showrows
```

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**